dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Basics/BooleanTable.mo

```diff
@@ -0,0 +1,49 @@
+within Dynawo.Electrical.Controls.Basics;
+
+/*
+* Copyright (c) 2022, RTE (http://www.rte-france.com)
+* See AUTHORS.txt
+* All rights reserved.
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite
+* of simulation tools for power systems.
+*/
+
+model BooleanTable "Generates a Boolean output signal based on a vector of time instants"
+  import Modelica;
+  import Dynawo;
+
+  parameter Modelica.Blocks.Types.Extrapolation Extrapolation = Modelica.Blocks.Types.Extrapolation.HoldLastPoint
+    "Extrapolation of data outside the definition range";
+  parameter String FileName = "NoName" "File where vector is stored"
+    annotation(Dialog(
+      enable=TableOnFile,
+      loadSelector(filter="Text files (*.txt);;MATLAB MAT-files (*.mat)",
+          caption="Open file in which table is present")));
+  parameter String TableName = "NoName"
+    "Table name on file or in function usertab (see docu)";
+  parameter Dynawo.Types.Time tShift = 0
```

Comment:
What does it do ?

Reply:
tShift moves the time abscissas in an uniform way. It may be useless given the ease with which the .txt can be modified.

Reply:
Yes, we can discard it

Reply:
Done.

Reply:
And tStart = startTime ? If time < tStart = -Modelica.Constants.inf, the output is false. In my view, tStart is useless too.

Reply:
Yes, we can too

Reply:
Deleted